infra/aws/terraform/prow-build-cluster/eks.tf

```diff
@@ -32,13 +32,41 @@ module "eks" {
 
   # Configure aws-auth
   aws_auth_roles = [
+    # Allow access to the Prow-EKS-Admin IAM role (used by Prow directly).
     {
       "rolearn"  = aws_iam_role.eks_admin.arn
       "username" = "eks-admin"
       "groups" = [
         "system:masters"
       ]
-    }
+    },
+    # Allow access to the Prow-Cluster-Admin IAM role (used with assume role with other IAM accounts).
+    {
+      "rolearn"  = aws_iam_role.iam_cluster_admin.arn
+      "username" = "eks-cluster-admin"
+      "groups" = [
+        "system:masters"
```

Comment:
Can we provide access less than `system:masters`?

Reply:
Similar to as Prow, we can eventually bind to some SA that's cluster-admin.